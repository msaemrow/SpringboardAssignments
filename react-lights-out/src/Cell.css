.Cell {
    height: 100px;
    width: 100px;
    background-color: darkgray;
    cursor: pointer;
    border: solid 1px black;
  }
  
  .Cell-lit {
    background-color: yellow;
  }