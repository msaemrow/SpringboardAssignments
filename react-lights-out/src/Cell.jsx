import React from "react";
import "./Cell.css";
/** A single cell on the board.
 *
 * This has no state --- just two props:
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 *
 * - isLit: boolean, is this cell lit?
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/


function Cell({  id, isLit, flipCellsAroundMe }) {
  const classes = `Cell ${isLit ? "Cell-lit" : ""}`;
  const handleClick = () => {
    if(typeof flipCellsAroundMe === 'function'){
        flipCellsAroundMe();
    }
  }
  return <>
    <td className={classes} id={ id } onClick={handleClick}/>
    </>
}

export default Cell;
