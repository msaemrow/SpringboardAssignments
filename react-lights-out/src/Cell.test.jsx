import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Cell from "./Cell"

describe('checks Cell componenet', () => {
    it('renders without crashing', () => {
        render(<Cell flipCellsAroundMe isLit={true}/>);
      });

  it('matches snapshot', () => {
    const { asFragment } = render(<Cell flipCellsAroundMe isLit={true}/>);
    expect(asFragment).toMatchSnapshot();
  });
});

describe('test', () => {
    it('checks truthy', () => {
        expect(4+4).toBe(8);
    })
})