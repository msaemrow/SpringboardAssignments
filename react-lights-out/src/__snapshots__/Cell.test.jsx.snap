// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`checks Cell componenet > matches snapshot 1`] = `[Function]`;
